clmm/theory/parent_class.py

```diff
@@ -426,7 +426,7 @@ def _eval_convergence(self, r_proj, z_cl, z_src):
         return sigma/sigma_c
 
     def eval_reduced_tangential_shear(self, r_proj, z_cl, z_src, z_src_model='single_plane',
-                                      beta_s_mean=None, beta_s_square_mean=None):
+                                      beta_s_mean=None, beta_s_square_mean=None, z_distrib_func=None):
```

Comment:
With the new non-single-plane options, should `z_src` be made an optional argument, with `None` a valid option? In that case, we won't have individual `z_src` values to provide.

Linked to that, and if agreed with, in the argument validation step (line 471-474):
- it should be extended to check the types of the new `beta` parameters;
- If `z_src` becomes an optional parameter, with `None` as a valid value, then the validation line should become `validate_argument(locals(), 'z_src', 'float_array', argmin=0, none_ok=True)`.

Reply:
I agree with making `z_src` an option variable. Maybe for the test it could be something like
````
if z_src_model=='single_plane':
    validate_argument(locals(), 'z_cl', float, argmin=0)
else:
    validate_argument(locals(), 'z_src', 'float_array', argmin=0, none_ok=True).
````
This way `None` won't be accepted if the function is called with the `single_plane` option.